j4-make-config

```diff
@@ -61,184 +57,182 @@ Append additional configuration and reload i3 after creating the config file:
-				windowActive = True
-				barActive = False
-			elif "$i3-theme-bar" in line:
-				windowActive = False
-				barActive = True
-			else:
-				if windowActive and not barActive:
-					windowText += line
-				elif not windowActive and barActive:
-					barText += line
-		themefile.close()
-	
-	# try to open i3 base config file
-	try:
-		basefile = open(i3_BASECONFIG_PATH, "r")
-	except IOError:
-		print("Error: i3 base config file could not be opened.")
-		exit(1)
-	# read all lines of base config into list
-	baselines_list.extend(basefile.readlines())
-	basefile.close()
-	
-	# append all additional configuration files
-	for appendFilename in appendFiles_list:
-		try:
-			appendFile = open(i3_PATH + "/" + appendFilename, "r")
-		except IOError:
-			print("Error: file %s could not be opened." % appendFilename)
-			exit(1)
-		# read all lines of appendFile into list
-		baselines_list.extend(appendFile.readlines())
-		appendFile.close()
-	
-	# try to open i3 config file
-	try:
-		configfile = open(i3_CONFIG_PATH, "w")
-	except IOError:
-		print("Error: i3 config file could not be created.")
-		exit(1)
-	
-	# iterate over base config lines, write all to config file
-	# and insert theme config at the right place
-	for line in baselines_list:
-		if line.lstrip().startswith("# $i3-theme-window"):
-			if not noTheme:
-				configfile.write(windowText)
-		elif line.lstrip().startswith("# $i3-theme-bar"):
-			if not noTheme:
-				configfile.write(barText)
-		else:
-			configfile.write(line)
-	configfile.flush()
-	configfile.close()
-
-	if doReload:
-		system("i3-msg reload")
-	
-	# store commandstring in rc file
-	if (argv[1] != "-i"):
-		commandString = ' '.join(argv)
-	else:
-		commandString = argv[0] + ' -r ' + theme_name
-	try:
-		rcfile = open(getenv("HOME") + "/.j4-make-config.rc", "w")
-	except IOError:
-		print("Error: file %s could not be opened." % (getenv("HOME") + "/.j4-make-config.rc"))
-		exit(1)
-	rcfile.write(commandString)
-	rcfile.flush()
-	rcfile.close()
-	
-	exit(0)
+    ## Configuring the work environment.
+    # detect i3 configuration directory
+    if not detectConfigDir():
+        print("Error: i3 base config file could not be found.")
+        exit(1)
+    else:
+        # set paths
+        THEME_PATH = i3_PATH + "/themes"
+        i3_BASECONFIG_PATH = i3_PATH + "/config.base"
+        i3_CONFIG_PATH = i3_PATH + "/config"
+        # fill themes dictionary with names and paths
+        try:
+            for theme in listdir(SYSTEM_THEME_PATH):
+                themes_list[theme] = SYSTEM_THEME_PATH + sep + theme
+        except (IOError, OSError):
+            pass
+
+        try:
+            for theme in listdir(THEME_PATH):
+                themes_list[theme] = THEME_PATH + sep + theme
+        except (IOError, OSError):
+            pass
+
+        if themes_list == {}:
+            print("Error: No themes found.")
+            exit(1)
+
+    parser = argparse.ArgumentParser(description='i3 config generator/switcher',
```

Comment:
`description` might be better left unset because if you looked in `--help`, you'll see two of them. The other one is in `UsageInfo()`. With `argparse` in this PR, maybe the options should be printed correctly instead of faking it?

Reply:
Good suggestion. As it stands I can't think of a good way to show the usage information; it feels like it'd be better for a man page instead of the command line help. I added a "--themes" argument so we can preserve that but a "--usage" argument is weird to me.

A pet peeve: I'd like the arguments to be ordered so that '--themes' isn't put as the second argument but that's kind of finicky and requires knowledge of `argparse` internals that I do not have right now.

Reply:
I sent you a PR. You can continue working on top of that... enhancing `j4-make-config` further. :-)

This PR might alleviate some of your issues with `j4-make-config`. With too many themes, the list of themes could be hard to parse so I append them all on same line + 8 themes equals an additional line. Not sure if it's a good idea, but you can see the differences.

With `--list-themes` (and any other arguments, really), argparse permits least unambiguous prefix so we can just use `j4-make-config --list` too.